### PR TITLE
fix(nix): fix utop tests

### PR DESCRIPTION
**Pull request by Alizter**
Nix was compressing the man pages for utop which meant the dune-package file was pointing to non-existant files. Overriding utop and disabling the compression of man pages fixes this issue.

We can also use this fix for other `(package)` dependencies that are problematic with nix.

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -221,13 +225,24 @@
                 mdx
                 odoc
                 ppx_expect
+
```
Comment (rgrinberg) [inline flake.nix:228]: did this sneak in accidentally?

**Comment by rgrinberg** [comment]
It seems like a decent workaround. The problem with it is that it requires every single package with manpages that is depended on with `(package ..)` to be modified this way.

I don't know enough about nix, but surely there's other instances it tries to modify build artifacts and has to modify some metadata as well. Can we make nix modify the dune-package file to use the gz filename?

**Comment by Alizter** [comment]
@rgrinberg That should be possible. I'll ask around.

We can make the overlays a bit easier to work with by separating them out of the flake. It's already gotten pretty big.

**Comment by anmonteiro** [comment]
Probably easy to change `buildDunePackage` itself. that's what we did for `--man-dir` and other stuff before.

**Comment by Alizter** [comment]
@anmonteiro Yes, I've spoken to Ulrik about supporting this in buildDunePackage. For the moment, this PR solves our own issues with the test suite.

**Review by anmonteiro — APPROVED** [review]
makes sense to do it here while it's not fixed upstream

**Pull request merged**